src/Configuration/ConfigurationFactory.php

```diff
@@ -130,4 +134,15 @@ public function create(
             $minCoveredMsi
         );
     }
+
+    private function retrieveMutators(array $schemaMutators, string $mutatorsInput): array
```

Comment:
could you please specify array types?

Reply:
it's `mixed[]`... hence `array`. It's our "messy" type coming from the configuration file hence if it was typescript:

```typescript
type Ignore = {
  ignore?: string[];
}

type Settings = {
  settings:? {
    [key: string]?: string;
  }
}

type MutatorConfig = boolean | (Ignore & Settings);
```

Reply:
Then we can have exactly the same DTO:

```php
final class MutatorConfig {
    public array<int, string>|null $ignore;

    public array<string, string>|null $settings
}
```

and then

```diff
- private function retrieveMutators(array $schemaMutators, string $mutatorsInput): array
+ /**
+ * @param bool|MutatorConfig $schemaMutators
+ * ...
+ */
+ private function retrieveMutators(array $schemaMutators, string $mutatorsInput): array
```

Would it work? This is definitely out of the scope of this PR, but if it would work, we will implement it as a separate ticket to remove all these not clear `array`s

Reply:
It's not exactly the same though but I think we can improve it indeed. Opened https://github.com/infection/infection/issues/882 for it